### PR TITLE
feat(cli): Added support for provider argument in `prisma init`

### PR DESCRIPTION
**Pull request by SuryaElavazhagan**
Addresses #2779 issue

### CODE REVIEW

File: src/packages/cli/src/Init.ts [hunk 1]
```diff
@@ -143,6 +144,11 @@ export class Init implements Command {
       url = args['--url']
     }
 
+    if (args['--provider'] && !args['--url']) {
+      provider = args['--provider']
+      url = `${provider}://johndoe:randompassword@localhost:5432/mydb?schema=public`
```
Comment (janpio) [inline src/packages/cli/src/Init.ts:149]: We could differentiate the default port and existance of the schema (only on Postgres) as well.

Comment (SuryaElavazhagan) [reply src/packages/cli/src/Init.ts:149]: Nice one. Will add them..

Comment (SuryaElavazhagan) [reply src/packages/cli/src/Init.ts]: Please correct me if I'm wrong. I think you can query postgres to check for the existence of the schema only if you have the right credentials. It's very very rare to have the same credentials. I don't think it's worth it. I'd love to hear your thoughts on this.

Comment (janpio) [reply src/packages/cli/src/Init.ts]: What I meant is that on MySQL you do not specify a `schema` as it does not do anything. Only on PostgreSQL you do.

Comment (SuryaElavazhagan) [reply src/packages/cli/src/Init.ts]: Ah, sorry. Got it.

File: src/packages/cli/src/Init.ts [hunk 2]
```diff
@@ -143,6 +167,11 @@ export class Init implements Command {
       url = args['--url']
     }
 
+    if (args['--provider'] && !args['--url']) {
```
Comment (Jolg42) [inline src/packages/cli/src/Init.ts:170]:
```suggestion
    } else if (args['--provider']) {
```
Minor suggestion: I think this could be better

**Review by Jolg42 — APPROVED** [review]
This looks great!

I think we could add mongodb later.

It's in "Early Access" for now, more info at https://www.prisma.io/docs/concepts/database-connectors/mongodb

And probably we can use this for the default url
```
mongodb://johndoe:randompassword@localhost:27017/mydb?authSource=admin
```

**Review by Jolg42 — CHANGES_REQUESTED** [review]
Actually, this implementation doesn't work for `sqlserver` which is JDBC style like 
```
sqlserver://localhost:5432;database=mydb;user=SA;password=randompassword;"
```
(https://www.prisma.io/docs/concepts/components/preview-features/sql-server/sql-server-connection-string)

File: src/packages/cli/src/Init.ts [hunk 3]
```diff
@@ -49,6 +49,29 @@ export const defaultEnv = (
...
+export const defaultURL = (
+  provider = 'postgresql',
+  port = defaultPort(provider),
+  schema = 'public',
+) => {
+  const schemaExists = provider === 'postgresql'
+  return `${provider}://johndoe:randompassword@localhost:${port}/mydb${
+    schemaExists ? `?schema=${schema}` : ''
+  }`
+}
```
Comment (Jolg42) [inline src/packages/cli/src/Init.ts:73]: We could also add a `sqlite` provider here to be complete I think, default could be `file:./dev.db`

https://www.prisma.io/docs/concepts/database-connectors/sqlite

Comment (SuryaElavazhagan) [reply src/packages/cli/src/Init.ts]: Silly me. Thanks. I've responded to your comments btw.

**Review by pantharshit00 — APPROVED** [review]

**Review by williamluke4 — APPROVED** [review]

**Comment by Jolg42** [comment]
Thanks! 💚 

I added mongodb and small changes/fixes in https://github.com/prisma/prisma/pull/7557

**Pull request merged**